name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -190,17 +190,21 @@ public ChartInterval getInverval(Client client, CurrencyConverter converter, Sec
                                     ? LocalDate.now()
                                     : tx.get(tx.size() - 1).getTransaction().getDateTime().toLocalDate());
                 case ALL:
-                    List<SecurityPrice> prices = security.getPricesIncludingLatest();
-
-                    if (prices.isEmpty())
-                        return new ChartInterval(now, now);
-                    else
-                        return new ChartInterval(prices.get(0).getDate(), prices.get(prices.size() - 1).getDate());
+                    return getAllPrices(now, security);
 
                 default:
                     throw new IllegalArgumentException();
             }
         }
+
+        private ChartInterval getAllPrices(LocalDate now, Security security)
```

Comment:
I don’t like the name `getAllPrices()` because it doesn’t actually get prices, it finds out for which interval prices are available.

Reply:
I agree with you. What would be a good name? ``determinIntervalFromPrices``?